libs/nav/src/tpspace/motion_planning_utils.cpp

```diff
@@ -136,6 +139,92 @@ void mrpt::nav::build_PTG_collision_grids(
 
 	}	// "else" recompute all PTG
 
+	//Clearance grid
+	if (PT->LoadCleGridsFromFile( cacheFilename, robotShape ) )
+	{
+		if (verbose)
+			cout << "ClearanceGrid loaded from file OK" << endl;
+	}
+	else
+	{
+		// BUGFIX: In case we start reading the file and in the end detected an error,
+		//         we must make sure that there's space enough for the grid:
+		PT->m_clearanceGrid.setSize( -PT->refDistance,PT->refDistance,-PT->refDistance,PT->refDistance,PT->m_clearanceGrid.getResolution());
+
+		const int grid_cx_max = PT->m_clearanceGrid.getSizeX()-1;
+		const int grid_cy_max = PT->m_clearanceGrid.getSizeY()-1;
+
+		const size_t nVerts = robotShape.verticesCount();
+		std::vector<mrpt::math::TPoint2D> transf_shape(nVerts); // The robot shape at each location
+
+		// RECOMPUTE THE COLLISION GRIDS:
+		// ---------------------------------------
+		for (size_t k=0;k<Ki;k++)
+		{
+			const size_t nPoints = PT->getPointsCountInCPath_k(k);
+			ASSERT_(nPoints>1)
+
+			for (size_t n=0;n<(nPoints-1);n++)
+			{
+				// Translate and rotate the robot shape at this C-Space pose:
+				const double x   = PT->GetCPathPoint_x( k,n );
+				const double y   = PT->GetCPathPoint_y( k,n );
+				const double phi = PT->GetCPathPoint_phi( k,n );
+
+				mrpt::math::TPoint2D bb_min(std::numeric_limits<double>::max(),std::numeric_limits<double>::max());
+				mrpt::math::TPoint2D bb_max(-std::numeric_limits<double>::max(),-std::numeric_limits<double>::max());
+
+				for (size_t m = 0;m<nVerts;m++)
+				{
+					transf_shape[m].x = x + cos(phi)*robotShape.GetVertex_x(m)-sin(phi)*robotShape.GetVertex_y(m);
+					transf_shape[m].y = y + sin(phi)*robotShape.GetVertex_x(m)+cos(phi)*robotShape.GetVertex_y(m);
+					mrpt::utils::keep_max( bb_max.x, transf_shape[m].x); mrpt::utils::keep_max( bb_max.y, transf_shape[m].y);
+					mrpt::utils::keep_min( bb_min.x, transf_shape[m].x); mrpt::utils::keep_min( bb_min.y, transf_shape[m].y);
+				}
+
+				// Robot shape polygon:
+				const mrpt::math::TPolygon2D poly(transf_shape);
+
+				// Get the range of cells that may collide with this shape:
+				const int ix_min = std::max(0,PT->m_clearanceGrid.x2idx(bb_min.x)-1);
+				const int iy_min = std::max(0,PT->m_clearanceGrid.y2idx(bb_min.y)-1);
+				const int ix_max = std::min(PT->m_clearanceGrid.x2idx(bb_max.x)+1,grid_cx_max);
+				const int iy_max = std::min(PT->m_clearanceGrid.y2idx(bb_max.y)+1,grid_cy_max);
+
```

Comment:
@chatrasen Here we should have to evalute distances for a larger range... this "+-1" was though for collisions, that is why the check does not extend further. 
How many cells? I think the best would be to offer a new parameter (input argument to this function) defining the maximum clearance in meters to be checked.... it's an idea to start with...